scipy/stats/_variation.py

```diff
@@ -181,48 +95,31 @@ def variation(a, axis=0, nan_policy='propagate', ddof=0, *, keepdims=False):
     array([1.05109361, 0.31428986, 0.146483  ])
 
     """
-    a, axis = _chk_asarray(a, axis)
-    axis = normalize_axis_index(axis, ndim=a.ndim)
+    # `nan_policy` and `keepdims` are handled by `_axis_nan_policy`
     n = a.shape[axis]
     NaN = _get_nan(a)
 
-    contains_nan, nan_policy = _contains_nan(a, nan_policy)
-    if contains_nan and nan_policy == 'omit':
-        return _nanvariation(a, axis=axis, ddof=ddof, keepdims=keepdims)
-
     if a.size == 0 or ddof > n:
         # Handle as a special case to avoid spurious warnings.
         # The return values, if any, are all nan.
-        shp = list(a.shape)
-        if keepdims:
-            shp[axis] = 1
-        else:
-            del shp[axis]
-        if len(shp) == 0:
-            result = NaN
-        else:
-            result = np.full(shp, fill_value=NaN)
-
+        shp = np.asarray(a.shape)
+        shp = np.delete(shp, axis)
```

Comment:
Allows for `axis` tuples. Maybe not needed because I think the decorator reduces axis tuples to a single axis. I forgot.